src/test/java/com/nvidia/spark/rapids/jni/CastStringsTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni;
+
+import ai.rapids.cudf.AssertUtils;
+import ai.rapids.cudf.ColumnVector;
+import ai.rapids.cudf.DType;
+import ai.rapids.cudf.Table;
+import org.junit.jupiter.api.Test;
+
+import java.math.RoundingMode;
+import java.util.stream.IntStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CastStringsTest {
+  @Test
+  void castToIntegerTest() {
+    Table.TestBuilder tb = new Table.TestBuilder();
+    tb.column(3l, 9l, 4l, 2l, 20l, null, null);
+    tb.column(5, 1, 0, 2, 7, null, null);
+    tb.column(new Byte[]{2, 3, 4, 5, 9, null, null});
+    Table expected = tb.build();
+
+    Table.TestBuilder tb2 = new Table.TestBuilder();
+    tb2.column("3", "9", "4", "2", "20", null, "7.6asd");
+    tb2.column("5", "1", "0", "2", "7", null, "asdf");
+    tb2.column("2", "3", "4", "5", "9", null, "7.8.3");
+
+    List<ColumnVector> result = new ArrayList<>();
+
+    try (Table origTable = tb2.build()) {
+      for (int i = 0; i < origTable.getNumberOfColumns(); i++) {
+        ColumnVector string_col = origTable.getColumn(i);
+        result.add(CastStrings.toInteger(string_col, false, 
+                   expected.getColumn(i).getType()));
+      }
+      Table result_tbl = new Table(
+        result.toArray(new ColumnVector[result.size()]));
+      AssertUtils.assertTablesAreEqual(expected, result_tbl);
+    }
+  }
+
+  @Test
+  void castToIntegerAnsiTest() {
+    Table.TestBuilder tb = new Table.TestBuilder();
+    tb.column(3l, 9l, 4l, 2l, 20l);
+    tb.column(5, 1, 0, 2, 7);
+    tb.column(new Byte[]{2, 3, 4, 5, 9});
+    Table expected = tb.build();
+
+    Table.TestBuilder tb2 = new Table.TestBuilder();
+    tb2.column("3", "9", "4", "2", "20");
+    tb2.column("5", "1", "0", "2", "7");
+    tb2.column("2", "3", "4", "5", "9");
```

Comment:
Do these values tell us anything about whether the ANSI flag is being passed properly?  Seems like these values would pass whether or not we properly pass `true` to the native code.

Reply:
Yeah, I was thinking I should test a failure, but I am not entirely sure how to catch the exception and validate it is the correct exception. I want to add that though, so I will play around with it later tonight or tomorrow.

Reply:
Couple of strategies:

If you just want to check that a specific exception type is thrown, `assertThrows` is appropriate, see an example at https://github.com/rapidsai/cudf/blob/branch-22.10/java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java#L847.

If you want to examine the exception (e.g.: check what's in the message or other extended checking) then this works:
```
try {
   do_the_operation();
   Assertions.fail("Should have thrown!");
} catch (ExpectedException e) {
   Assertions.assertTrue(e.getMessage().contains("some expected message string")
}
```